Feat: Space Battle rework

ItsMeJaba opened the pull request:
# Описание
Изначальный план был в ребалансе объема лута, что выносят из данного гейта со способностью снарядить всю свою братию культистов/революционеров и т.п. Но внешний вид и наполнение данного гейта меня совсем не устроил, поэтому я выкатываю данный реворк.

# Ссылка на предложение/Причина создания ПР
Оповещение: https://discord.com/channels/617003227182792704/930740675350233098/1081935077921009704
V2: https://discord.com/channels/617003227182792704/1084191571815182446
A-vote: https://discord.com/channels/617003227182792704/1078286061140193300/1091344748092063825
Отчет: https://discord.com/channels/617003227182792704/755125334097133628/1092568308106133585

# Демонстрация изменений
## Было:
![2023 03 01-00 16 33](https://user-images.githubusercontent.com/91949115/222921905-60944059-a9a9-42ad-ab13-b2aaea83e7cb.png)
## Стало:
![2023 03 04-22 57 38](https://user-images.githubusercontent.com/91949115/222921914-225f4b0f-d7b8-445d-be6f-364c13a3e458.png)

# Что изменилось:

-  Ребаланс лута: добавлена рандомизация содержимого карманов и сумок некоторых трупов. Противники теперь не оставляют тело после себя, рассыпаясь в пыль и так же с некоторым шансом могут оставить соответствующие предметы:
_Указаны те, что достаточно влияют на баланс и являются самыми неприятными для выноса_
   * Корабельные офицеры (СБ) - вооружение: 70% энфорсер+магазин, 20% ВТ+магазин, 10% дробовик+коробка бакшотов
   * Все синдикатовцы (Самое неприятное, что может попасть на станцию):
           2% - Сигареты синдиката, синдидонккпокет, ID-карта синдиката
    * Мобы в бладригах - 5% на выпадение хардсьюта
    * Дальние мобы - 1% C20r, 3% обойма C20r, 5% стечкин (патроны выпадают от 1 до 3 достаточно часто)
    * Ближние мобы - 1% есворд, 2% энергощит
    * Дроны синдиката - 100% выпадение одной платы с любой(кроме абдукторов) технологией уровня 1-4, синдикатовские технологии максимум 3

-  Добавлены маркеры-спавнеры (выглядят как мины с восклицательным знаком на рендере)
     Привязанные id маркеры спавнят на маркерах с изображением противника новых мобов, что добавляет немного больше динамики. На время тестирования маркеры вызывают у игрока над головой восклицательный знак как если бы он сидел в коробке и её открыли (отсылка на MGS), также по выходу из тестирования спрайты будут перенесены к другим спавнерам, что готовит Момо.

 - Ребаланс противников
   * Уровень здоровья снижен с 300-400 до адекватных 100 у всех синдикатовцев, кроме "босса" на мостике.
   * Т.к. шансы на выпадения серьезного лута и максимальная планка здоровья снизились, произошла компенсация количеством: (Было/стало/может заспавнится на маркерах)
     Босс-командир 1/1/0
     Ближний бой 23/25/12
     Дальний бой 18/44/7
     Ближний в бладриге 8/8/12
     Дальний в бладриге 21/37/30
     Дроны синдиката 0/76/69

- Переработка основного крейсера
   * Изменение общего оснащение помещений
   * Добавлен единственный ценный риг - СБ
   * Добавлена возможность собрать под
   * Удалены все спавнеры лута уровня дигла по 50 брута
   * Все ПНВ заменены на мезоны, единственные ПНВ на корабле - шлемы, что дают меньше обзора, чем фонарик
   * Добавлены записки: предсмертная механика, офицера в туалете; распоряжения НТ на мостике; фрагмент бортового журнала.
   * Бортовые турели теперь зависят от питания
  
- Переработка всех трех кораблей синдиката
   * Левый(имеет точку появления): гарантированный бладриг, акселераторную пушку, ПНВ
   * Центральный: гарантированный бладриг, пару лазерных винтовок, гарантировнный есворд(грифски)
   * Правый: шкафомех, 7 мимиков(серые ящики) 

- Добавлен корабль-якорь у двигателя крейсера
  *  Гарантированный атмос голопроектор, приличные объемы плазмы

- Карго-корабль (у самого правого  края)
  * Abandoned crate
  * Кларк + бур, гидравлическая клешня

- Амогусы
  * Нижний удален
  * Верхний стал захоронением hive-ботов, большая часть находится по периметру внутри камня.

 - Метеориты и другие булдыги на карте
  * Значительно снижено количество камня с повышенным шансом руды
  * Постарался снизить шансы спавна пещер с грибами за счет установки других типов камня
  
- Справа снизу добавлена отсылка - медведь Винни

ItsMeJaba committed:
Прошу прощения за мой говнокод

ItsMeJaba committed:
Fix: Blackmarket

Замена пути с измененного моба на альтернативу с сохранением характеристик

ItsMeJaba committed:
Попытка в фиксы №1

Починил то, на что ругается бот

Den109G commented:
СМище.... Надеюсь оно не бахает сразу.

ItsMeJaba committed:
Правки

Исправлено выпадение лута с дронов и вообще лута с мобов. CL бот идет лесом, /var/ нужен для работы моего говнокода

Небольшой ребаланс после тестирования
Фикс мелкой хрени

Herobrine998 commented:
Автор же понимает, что если реворк не пройдет голосовач, то он потратит время в пустую?

ItsMeJaba committed:
Доработки

+4 записки, милишники бьют берном, небольшой ребаланс по луту, похеренные гарантированные бладриги

ItsMeJaba committed:
Фиксы

Фикс после выпила req_access_txt на req_access = list(). Небольшие дофиксы.

ItsMeJaba committed:
Фиксы

Фикс после выпила req_access_txt на req_access = list(). Небольшие дофиксы.

Legendaxe commented on code/modules/awaymissions/mission_code/spacebattle.dm:
```diff
@@ -131,62 +136,578 @@
+	/var/synmobdrop //Обычный лут, дропается со всех
+	/var/SynSpace //Выпадение бладрига
+	/var/SynMelee //Лут с милишников
+	/var/SynRange //Лут с дальников
```
```suggestion
	var/SynRange //Лут с дальников
```

Legendaxe commented on code/modules/awaymissions/mission_code/spacebattle.dm:
```diff
@@ -131,62 +136,578 @@
+/mob/living/simple_animal/hostile/syndicate
+	/var/synmobdrop //Обычный лут, дропается со всех
+	/var/SynSpace //Выпадение бладрига
+	/var/SynMelee //Лут с милишников
```
```suggestion
	var/SynMelee //Лут с милишников
```

Legendaxe commented on code/modules/awaymissions/mission_code/spacebattle.dm:
```diff
@@ -131,62 +136,578 @@
+
+/mob/living/simple_animal/hostile/syndicate
+	/var/synmobdrop //Обычный лут, дропается со всех
+	/var/SynSpace //Выпадение бладрига
```
```suggestion
	var/SynSpace //Выпадение бладрига
```

Legendaxe commented on code/modules/awaymissions/mission_code/spacebattle.dm:
```diff
@@ -131,62 +136,578 @@
+// Enemys
+
+/mob/living/simple_animal/hostile/syndicate
+	/var/synmobdrop //Обычный лут, дропается со всех
```
```suggestion
	var/synmobdrop //Обычный лут, дропается со всех
```

Legendaxe commented on code/modules/awaymissions/mission_code/spacebattle.dm:
```diff
@@ -131,62 +136,578 @@
+	attack_sound = 'sound/weapons/saberon.ogg'
+
+/mob/living/simple_animal/hostile/syndicate/melee/autogib/spacebattle/Initialize()
+	loot = list(/obj/effect/decal/cleanable/ash, synmobdrop, SynMelee)
```
```suggestion
	. = ..()
	loot = list(/obj/effect/decal/cleanable/ash, synmobdrop, SynMelee)
```

Legendaxe commented on code/modules/awaymissions/mission_code/spacebattle.dm:
```diff
@@ -131,62 +136,578 @@
+
+/mob/living/simple_animal/hostile/syndicate/melee/autogib/spacebattle/Initialize()
+	loot = list(/obj/effect/decal/cleanable/ash, synmobdrop, SynMelee)
+	return ..()
```
```suggestion
	return .
```

Legendaxe commented on code/modules/awaymissions/mission_code/spacebattle.dm:
```diff
@@ -131,62 +136,578 @@
+	attack_sound = 'sound/weapons/saberon.ogg'
+
+/mob/living/simple_animal/hostile/syndicate/melee/space/autogib/spacebattle/Initialize()
+	loot = list(/obj/effect/decal/cleanable/ash, synmobdrop, SynMelee, SynSpace)
```
```suggestion
	. = ..()
	loot = list(/obj/effect/decal/cleanable/ash, synmobdrop, SynMelee, SynSpace)
```

Legendaxe commented on code/modules/awaymissions/mission_code/spacebattle.dm:
```diff
@@ -131,62 +136,578 @@
+
+/mob/living/simple_animal/hostile/syndicate/melee/space/autogib/spacebattle/Initialize()
+	loot = list(/obj/effect/decal/cleanable/ash, synmobdrop, SynMelee, SynSpace)
+	return ..()
```
```suggestion
	return .
```

Legendaxe commented on code/modules/awaymissions/mission_code/spacebattle.dm:
```diff
@@ -131,62 +136,578 @@
+	damage_coeff = list("brute" = 1, "fire" = 1, "tox" = 1, "clone" = 2, "stamina" = 0, "oxy" = 0.5)
+
+/mob/living/simple_animal/hostile/syndicate/ranged/autogib/spacebattle/Initialize()
+	loot = list(/obj/effect/decal/cleanable/ash, synmobdrop, SynRange)
```
```suggestion
	. = ..()
	loot = list(/obj/effect/decal/cleanable/ash, synmobdrop, SynRange)
```

Legendaxe commented on code/modules/awaymissions/mission_code/spacebattle.dm:
```diff
@@ -131,62 +136,578 @@
+
+/mob/living/simple_animal/hostile/syndicate/ranged/autogib/spacebattle/Initialize()
+	loot = list(/obj/effect/decal/cleanable/ash, synmobdrop, SynRange)
+	return ..()
```
```suggestion
	return .
```

Legendaxe commented on code/modules/awaymissions/mission_code/spacebattle.dm:
```diff
@@ -131,62 +136,578 @@
+/mob/living/simple_animal/hostile/syndicate/ranged/space/autogib/spacebattle
+
+/mob/living/simple_animal/hostile/syndicate/ranged/space/autogib/spacebattle/Initialize()
+	loot = list(/obj/effect/decal/cleanable/ash, synmobdrop, SynRange, SynSpace)
```
```suggestion
	. = ..()
	loot = list(/obj/effect/decal/cleanable/ash, synmobdrop, SynRange, SynSpace)
```

Legendaxe commented on code/modules/awaymissions/mission_code/spacebattle.dm:
```diff
@@ -131,62 +136,578 @@
+
+/mob/living/simple_animal/hostile/syndicate/ranged/space/autogib/spacebattle/Initialize()
+	loot = list(/obj/effect/decal/cleanable/ash, synmobdrop, SynRange, SynSpace)
+	return ..()
```
```suggestion
	return .
```

Legendaxe reviewed (changes requested):
Вызов родительского Initialize необходим для того что бы твои переменные с потенциальным лутом были обработаны и заполнены перед тем как ты пытаешься получить из них данные.

ItsMeJaba committed:
Apply suggestions from code review

Co-authored-by: Volodymir Ohorodnytskyi <102746941+Legendaxe@users.noreply.github.com>

ItsMeJaba committed:
Merge branch 'NSB' of https://github.com/ItsMeJaba/Paradise into NSB
